exo-go/src/service_restarter/index.go

```diff
@@ -0,0 +1,79 @@
+package serviceRestarter
+
+import (
+	"fmt"
+
+	"github.com/Originate/exosphere/exo-go/src/docker_compose"
+	"github.com/fsnotify/fsnotify"
+)
+
+// ServiceRestarter watches the given local service for changes and restarts it
+type ServiceRestarter struct {
+	ServiceName      string
+	ServiceDir       string
+	DockerComposeDir string
+	Env              []string
+	Log              func(string)
+	watcher          *fsnotify.Watcher
+}
+
+// Watch watches the service directory for changes
+func (s *ServiceRestarter) Watch(watcherErr chan<- error) error {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return err
+	}
+	s.watcher = watcher
+	go func() {
+		for {
+			select {
+			case event := <-watcher.Events:
+				if isCreate(event) {
+					s.Log(fmt.Sprintf("Restarting service '%s' because %s was created", s.ServiceName, event.Name))
+				} else if isRemove(event) {
+					s.Log(fmt.Sprintf("Restarting service '%s' because %s was deleted", s.ServiceName, event.Name))
+				} else if isChange(event) {
+					s.Log(fmt.Sprintf("Restarting service '%s' because %s was changed", s.ServiceName, event.Name))
+				}
+				watcherErr <- s.restart(watcherErr)
```

Comment:
I don't think we should push nil into the error channel. I think we should only push errors.